File: src/main/java/nl/hand/made/vending/machine/exception/ValidationExceptionHandler.java
package nl.hand.made.vending.machine.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class ValidationExceptionHandler {

    @ExceptionHandler(ValidationException.class)
    public final ResponseEntity<String> handleAllExceptions(ValidationException ex) {
        return new ResponseEntity<>(ex.getMessage(), HttpStatus.BAD_REQUEST);
    }
}

File: src/main/kotlin/nl/hand/made/vending/machine/exception/ValidationExceptionHandler.kt
package nl.hand.made.vending.machine.exception

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class ValidationExceptionHandler {

    @ExceptionHandler(ValidationException::class)
    fun handleAllExceptions(ex: ValidationException) = ResponseEntity(ex.message, HttpStatus.BAD_REQUEST)
}
